graph/src/util/log.rs

```diff
@@ -69,5 +74,15 @@ pub fn register_panic_hook(panic_logger: slog::Logger) {
                 );
             }
         };
+
+        if let Ok(ref mut mutex) = shutdown_sender.lock() {
```

Comment:
The mutex may be unwrapped, since it will only error if a panic happens while holding the lock, but in this case that would be a double panic which will abort.